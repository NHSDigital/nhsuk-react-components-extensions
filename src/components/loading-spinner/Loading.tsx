import React, { HTMLProps } from 'react';
import { Panel } from 'nhsuk-react-components';

interface Props extends HTMLProps<HTMLDivElement> {
  text?: string | false;
  backdrop?: boolean;
  shown?: boolean;
}

const Loading: React.FC<Props> = ({ backdrop, shown, text, width, height }) => {
  if (!shown) return null;

  const baseSpinner = (
    <Panel className="nhsuk-loader__panel" style={{ minWidth: width, minHeight: height }}>
      <div style={{ width, minHeight: height }} className="nhsuk-loader" />
      {text && <p className="nhsuk-heading-l nhsuk-loader__text">{text}</p>}
    </Panel>
  );

  return backdrop ? <div className="nhsuk-loader__backdrop">{baseSpinner}</div> : baseSpinner;
};

Loading.defaultProps = {
  text: 'Loading...',
  width: 200,
  height: 200,
  shown: true,
  backdrop: true,
};

export default Loading;
