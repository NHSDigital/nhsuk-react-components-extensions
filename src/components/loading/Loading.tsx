import React, { HTMLProps } from 'react';

interface Props extends HTMLProps<HTMLDivElement> {
  text?: string;
}

const Loading: React.FC<Props> = ({ text, ...rest }: Props) => (
  <div {...rest}>
    <div className="nhsuk-loader__container">
      <p className="nhsuk-heading-l">{text}</p>
    </div>

    <div className="nhsuk-loader__container">
      <div className="nhsuk-loader" />
    </div>
  </div>
);

Loading.defaultProps = {
  text: 'Loading...',
};

export default Loading;
