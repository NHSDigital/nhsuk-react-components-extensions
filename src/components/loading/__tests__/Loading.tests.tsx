import Loading from '..';
import React from 'react';
import { shallow } from 'enzyme';

describe('loading', () => {
  it('renders correctly given text', () => {
    const component = shallow(
      <Loading text="I am testing the loading component" />
    );
    expect(component).toMatchSnapshot();
  });

  it('renders with default text if none supplied', () => {
    const component = shallow(<Loading />);
    expect(component).toMatchSnapshot();
  });
});
