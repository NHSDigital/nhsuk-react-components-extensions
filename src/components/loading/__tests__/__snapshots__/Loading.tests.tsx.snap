// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading renders correctly given text 1`] = `
<div>
  <div
    className="nhsuk-loader__container"
  >
    <p
      className="nhsuk-heading-l"
    >
      I am testing the loading component
    </p>
  </div>
  <div
    className="nhsuk-loader__container"
  >
    <div
      className="nhsuk-loader"
    />
  </div>
</div>
`;

exports[`loading renders with default text if none supplied 1`] = `
<div>
  <div
    className="nhsuk-loader__container"
  >
    <p
      className="nhsuk-heading-l"
    >
      Loading...
    </p>
  </div>
  <div
    className="nhsuk-loader__container"
  >
    <div
      className="nhsuk-loader"
    />
  </div>
</div>
`;
